databricks_cli/unity_catalog/uc_service.py

```diff
@@ -0,0 +1,448 @@
+# Databricks CLI
+# Copyright 2021 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from databricks_cli.unity_catalog.utils import mc_pretty_format
+
+
+class UnityCatalogService(object):
+    def __init__(self, client):
+        self.client = client
+
+    # Metastore Operations
+
+    def create_metastore(self, name, storage_root, headers=None):
+        _data = {
+            'name': name,
+            'storage_root': storage_root,
+        }
+        return self.client.perform_query('POST', '/unity-catalog/metastores', data=_data,
+                                         headers=headers)
+
+    def list_metastores(self, headers=None):
+        _data = {}
+
+        return self.client.perform_query('GET', '/unity-catalog/metastores', data=_data,
+                                         headers=headers)
+
+    def get_metastore(self, metastore_id, headers=None):
+        _data = {}
+
+        return self.client.perform_query('GET', '/unity-catalog/metastores/%s' % (metastore_id),
```

Comment:
nit: should rather use f-strings, but then we finally need to drop python2 support - f'/unity-catalog/metastores/{metastore_id}'

Reply:
I've been holding off on using f-strings to maintain Python2 compatibility (until I hear solid evidence that we really can require Python3 going forward).